src/util/namespace.h

```diff
@@ -84,8 +84,20 @@ class namespacet:public namespace_baset
 
   using namespace_baset::lookup;
 
-  // these do the actual lookup
+  /// Searches for a symbol named \p name in `symbol_table1` and returns a
+  /// pointer to it in output parameter \p symbol when found. When no symbol is
+  /// found, it performs the same lookup in `symbol_table2`. If a symbol is
+  /// found it will be returned in \p symbol.
+  /// \return False iff the requested symbol is found in at least one of the two
+  /// tables
   virtual bool lookup(const irep_idt &name, const symbolt *&symbol) const;
+
+  /// Return the maximum integer x such that there is a symbol (in some of the
+  /// two symbol tables) whose name has \p prefix as prefix and x as suffix.
+  /// The intended use case is finding the next available symbol name for a
+  /// sequence of auto-generated symbols.
+  /// This method assumes that all symbols whose name starts with \p prefix have
+  /// a name of the form "prefixNNN" where NNN is an integer.
```

Comment:
I'm not sure which comment you might be referring to - mind providing a link? From my point of view the "fix" is to not include that sentence at all.